### PR TITLE
Event

### PR DESCRIPTION
**Pull request by jeaniewu**

### CODE REVIEW

**Comment by jayfriso** [comment]
:+1:

File: Assets/Scripts/Controller2.cs [hunk 1]
```diff
@@ -69,6 +70,25 @@ public class Controller2 : MonoBehaviour {
...
+		} else {
+			if (Event.current.keyCode == KeyCode.Z){
+				ghostMode = false;
+			}
+			if (Event.current.keyCode == KeyCode.X) {
+				levelManager.GetComponent<GhostSwitchManager> ().possessCurrentSelection ();
+			}
+		}
+		Event.current.Use ();
+	}
```
Comment (Gowtham100) [inline Assets/Scripts/Controller2.cs]: :+1:

File: Assets/Scripts/Controller2.cs [hunk 2]
```diff
@@ -70,62 +61,35 @@ public class Controller2 : MonoBehaviour {
...
-			if (Event.current.keyCode == KeyCode.Z){
+			if (Input.GetButtonDown("cancelGhostMode")){
 				ghostMode = false;
+				Input.ResetInputAxes();
 			}
-			if (Event.current.keyCode == KeyCode.X) {
+			if (Input.GetButtonDown ("Possess")) {
 				levelManager.GetComponent<GhostSwitchManager> ().possessCurrentSelection ();
 			}
 		}
```
Comment (Gowtham100) [inline Assets/Scripts/Controller2.cs:94]: :+1:

**Comment by Gowtham100** [comment]
:+1:

**Comment by jeaniewu** [comment]
created a separate task to update UI
https://trello.com/c/MRvIdbqg/153-update-control-tutorial-ui-after-the-new-control-changes

**Pull request merged**